src/NuGet.Core/NuGet.Commands/RestoreCommand/ContentFiles/VirtualFileInfo.cs

```diff
@@ -59,11 +59,24 @@ public long Length
             }
         }
 
+        private string _name;
         public string Name
         {
             get
             {
-                return PhysicalPath.Split('/').LastOrDefault();
+                if (_name == null)
+                {
+                    int lastSlashIndex = PhysicalPath.LastIndexOf('/');
+                    if (lastSlashIndex >= 0)
+                    {
+                        _name = PhysicalPath.Substring(lastSlashIndex + 1);
+                    }
+                    else
+                    {
+                        _name = PhysicalPath;
+                    }
```

Comment:
Perhaps we can use the [Path.GetFileName](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getfilename?view=net-8.0#system-io-path-getfilename(system-string)) method, which retrieves the file name and extension from a given path - this is exactly what we need to do. Please let me know if my understanding is incorrect.

Reply:
`Path.GetFileName` will also use `\` as a separator, at least on Windows. Whereas currently only `/` is used. Therefore it would be a change in behaviour, and it might cause NuGet on Windows to work differently to Linux/Mac (when a package was "incorrectly" packed with Windows `\` separators rather than `/` as pack normalizes to).

It's an edge case, but these 9 lines of code aren't so complex that I feel it's worthwhile investigating this more deeply to understand whether it will change any behaviour.